apps/mobile-wallet/src/persistent-storage/utils.ts

```diff
@@ -16,38 +16,86 @@ You should have received a copy of the GNU Lesser General Public License
 import AsyncStorage from '@react-native-async-storage/async-storage'
 import * as SecureStore from 'expo-secure-store'
 
 import { sendAnalytics } from '~/analytics'
 import { defaultSecureStoreConfig } from '~/persistent-storage/config'
 
-export const storeWithReportableError = async (key: string, value: string, error: string) => {
+export const getWithReportableError = async (key: string) => {
+  try {
+    return await AsyncStorage.getItem(key)
+  } catch (error) {
+    sendAnalytics({ type: 'error', error, message: `Could not get ${key} from AsyncStorage` })
+    throw error
+  }
+}
+
+export const storeWithReportableError = async (key: string, value: string) => {
   try {
     await AsyncStorage.setItem(key, value)
-  } catch (e) {
-    sendAnalytics('Error', { message: getHumanReadableError(e, error) })
-    throw new Error(error)
+  } catch (error) {
+    sendAnalytics({ type: 'error', error, message: `Could not store ${key} in AsyncStorage` })
+    throw error
+  }
+}
+
+export const deleteWithReportableError = async (key: string) => {
+  try {
+    await AsyncStorage.removeItem(key)
+  } catch (error) {
+    sendAnalytics({ type: 'error', error, message: `Could not delete ${key} from AsyncStorage` })
+    throw error
+  }
+}
+
+export const getSecurelyWithReportableError = async (
+  key: string,
+  showDefaultErrorMessage: boolean,
+  errorMessage: string
+) => {
+  try {
+    return await SecureStore.getItemAsync(key, defaultSecureStoreConfig)
+  } catch (error) {
+    sendAnalytics({
+      type: 'error',
+      message: showDefaultErrorMessage ? `Could not get ${key} from secure storage` : errorMessage
+    })
+    throw error
   }
 }
 
-export const storeSecurelyWithReportableError = async (key: string, value: string, error: string) => {
+export const storeSecurelyWithReportableError = async (
+  key: string,
+  value: string,
+  showDefaultErrorMessage: boolean,
+  errorMessage: string
+) => {
   try {
     await SecureStore.setItemAsync(key, value, defaultSecureStoreConfig)
-  } catch (e) {
-    sendAnalytics('Error', { message: error })
-    throw new Error(error)
+  } catch (error) {
+    sendAnalytics({
+      type: 'error',
+      message: showDefaultErrorMessage ? `Could not store ${key} in secure storage` : errorMessage
+    })
+    throw error
   } finally {
     value = ''
   }
 }
 
-export const deleteSecurelyWithReportableError = async (key: string, error: string) => {
+export const deleteSecurelyWithReportableError = async (
+  key: string,
+  showDefaultErrorMessage: boolean,
+  errorMessage: string
+) => {
   try {
-    await SecureStore.deleteItemAsync(key)
-  } catch (e) {
-    sendAnalytics('Error', { message: error })
-    throw new Error(error)
+    await SecureStore.deleteItemAsync(key, defaultSecureStoreConfig)
+  } catch (error) {
+    sendAnalytics({
+      type: 'error',
+      message: showDefaultErrorMessage ? `Could not delete ${key} from secure storage` : errorMessage
+    })
+    throw error
   }
 }
```

Comment:
This was done by design, I wanted the reportable error message to be an explicit thing. I don't want it to have a default because some storage keys might contain sensitive info, like the key of the private key entry (which contains the address hash). That's why I made the error message a mandatory arg, but also introduced the boolean arg `showDefaultErrorMessage` so that the developer will first check the implementation of the function before calling it. This was done to avoid leaking any unwanted sensitive info by accident. I favored privacy/security over code prettiness here.

Reply:
Makes sense. Thanks for the explanations 🙏